vm/src/lib.rs

```diff
@@ -7,7 +7,6 @@
 
 #![cfg_attr(docsrs, feature(doc_cfg))]
 #![deny(warnings)]
-#![forbid(unsafe_code)]
```

Comment:
Would be great to add a comment somewhere to explain why we allow unsafe code. also, is there a way to allow it only on places we need and not at the crate level ?